### PR TITLE
NUX & Domains: Auto Add Privacy Protection if plan is bought

### PR DESCRIPTION
**Pull request by aidvu**
Auto add privacy protection if a non-free plan is purchased during NUX

### CODE REVIEW

**Comment by aidvu** [comment]
@bisko Here goes. :) Should cover all! ALL!

File: client/signup/steps/plans/index.jsx [hunk 1]
```diff
@@ -28,8 +29,14 @@ class PlansStep extends Component {
 			stepSectionName,
 			stepName,
 			goToNextStep,
-			translate
-		} = this.props;
+			translate,
+			signupDependencies: { domainItem }
+		} = this.props,
+			privacyItem = isEmpty( cartItem )
```
Comment (umurkontaci) [inline client/signup/steps/plans/index.jsx]: `privacyItem = isEmpty( cartItem ) && cartItems.domainPrivacyProtection( {domain: domainItem.meta } );`

**Comment by aidvu** [comment]
Fixed #8614

**Comment by umurkontaci** [comment]
> Fixed #8614

In the commit message :)

**Comment by aidvu** [comment]
:sob:

File: client/signup/steps/plans/index.jsx (hunk 1 above)
Comment (aidvu) [inline client/signup/steps/plans/index.jsx]: Forgot the `! isEmpty()`.

File: client/signup/steps/plans/index.jsx [hunk 2]
```diff
@@ -28,8 +29,10 @@ class PlansStep extends Component {
 			stepSectionName,
 			stepName,
 			goToNextStep,
-			translate
-		} = this.props;
+			translate,
+			signupDependencies: { domainItem }
+		} = this.props,
+			privacyItem = ! isEmpty( cartItem ) && cartItems.domainPrivacyProtection( { domain: domainItem.meta } );
```
Comment (bisko) [inline client/signup/steps/plans/index.jsx]: This breaks if you don't have a domain selected:

`index.jsx:96 Uncaught TypeError: Cannot read property 'meta' of undefined`

To reproduce: Go through Signup and select a free `.wordpress.com` domain, but try to choose a paid plan after that.

**Comment by aidvu** [comment]
Nice. Thanks!

**Comment by aidvu** [comment]
@bisko Ready now. All cases should work.

**Pull request merged**